test/converters/xcresult/xcresult.go

```diff
@@ -0,0 +1,188 @@
+package xcresult
+
+import (
+	"encoding/xml"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"howett.net/plist"
+
+	"github.com/bitrise-io/go-utils/fileutil"
+)
+
+//
+// TestSummaries.plist
+
+type TestSummaryPlist struct {
+	FormatVersion     string
+	TestableSummaries []TestableSummary
+}
+
+func collapseSubtestTree(data Subtests) (tests Subtests) {
+	for _, test := range data {
+		if len(test.Subtests) > 0 {
+			tests = append(tests, collapseSubtestTree(test.Subtests)...)
+		}
+		if test.TestStatus != "" {
+			tests = append(tests, test)
+		}
+	}
+	return
+}
+
+func (summaryPlist TestSummaryPlist) Tests() map[string]Subtests {
+	tests := map[string]Subtests{}
+	var subTests Subtests
+	for _, testableSummary := range summaryPlist.TestableSummaries {
+		for _, test := range testableSummary.Tests {
+			subTests = append(subTests, collapseSubtestTree(test.Subtests)...)
+		}
+	}
+	for _, test := range subTests {
+		testID := strings.Split(test.TestIdentifier, "/")[0]
+		tests[testID] = append(tests[testID], test)
+	}
+	return tests
+}
+
+type TestableSummary struct {
+	TargetName      string
+	TestKind        string
+	TestName        string
+	TestObjectClass string
+	Tests           []Test
```

Comment:
can you move the Test/Tests definition closer?

Reply:
right below TestSummaryPlist, there is the only one reference for this model

Reply:
👍